### PR TITLE
[PREVIEW] DIV-3268 check your answer

### PR DESCRIPTION
**Pull request by hosainnet**
# Description

*  I can check my answers before submitting
GIVEN I am on the 'respond' page
WHEN I select a response and press continue
THEN I see the Check your answers page with my response correctly shown

* I can check my answers before submitting
GIVEN I am on the 'respond' page
WHEN I select a response and press continue
THEN I see the Check your answers page with my response correctly shown

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
   ],
   "dependencies": {
     "@hmcts/div-idam-express-middleware": "https://github.com/hmcts/div-idam-express-middleware#6.0.0",
-    "@hmcts/look-and-feel": "^1.18.2",
+    "@hmcts/look-and-feel": "https://github.com/hmcts/look-and-feel#6a4d5cd7f4710f4e0cb53905aa782c458862afc9",
```
Comment (hosainnet) [inline package.json]: temporary, pending publishing of 1.19.2

File: package.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
   ],
   "dependencies": {
     "@hmcts/div-idam-express-middleware": "https://github.com/hmcts/div-idam-express-middleware#6.0.0",
-    "@hmcts/look-and-feel": "^1.18.2",
+    "@hmcts/look-and-feel": "1.19.2",
```
Comment (hosainnet) [inline package.json]: The new version includes support for description of choices

File: steps/choose-a-response/ChooseAResponse.html [hunk 3]
```diff
@@ -1,64 +1,45 @@
...
-                <label class="form-label-bold" for="response-disagree">
-                    {{ content.fields.disagree.heading | safe }}<br>
-                    <span class="font-small">
-                    {{ content.fields.disagree.summary | safe }}
-                </span>
-                </label>
+                <h2 class="heading-small">{{ content.info.options.disagree.heading | safe }}</h2>
+                <p class="text">{{ content.info.options.disagree.summary | safe }}</p>
             </div>
-        </fieldset>
```
Comment (hosainnet) [inline steps/choose-a-response/ChooseAResponse.html]: Refactored the response choices to use `look-and-feel` macros, so we can get all the built-in functionality (remembering choices, error messages etc)

File: steps/choose-a-response/ChooseAResponse.step.js [hunk 4]
```diff
@@ -22,12 +27,20 @@ class ChooseAResponse extends Question {
...
   }
 
+  answers() {
+    const question = content.en.title;
+    return answer(this, {
+      question,
+      answer: this.fields.response.value === proceed ? content.en.fields.proceed.answer : content.en.fields.disagree.answer
+    });
+  }
+
```
Comment (hosainnet) [inline steps/choose-a-response/ChooseAResponse.step.js]: Overriding `answers()` allows us to tailor the message displayed in the next page (check-your-answers) as it's not exactly the same string on both pages

File: steps/respond/Respond.step.js [hunk 5]
```diff
@@ -13,6 +13,11 @@ class Respond extends Interstitial {
     return this.req.session;
   }
 
+  handler(req, res) {
+    req.session.entryPoint = this.name;
+    super.handler(req, res);
+  }
+
```
Comment (hosainnet) [inline steps/respond/Respond.step.js]: Required for the new check-your-answers page

File: steps/choose-a-response/ChooseAResponse.step.js [hunk 6]
```diff
@@ -22,12 +27,20 @@ class ChooseAResponse extends Question {
     return form({ response });
   }
 
+  answers() {
+    const question = content.en.title;
+    return answer(this, {
+      question,
+      answer: this.fields.response.value === proceed ? content.en.fields.proceed.answer : content.en.fields.disagree.answer
```
[the remark this answers is not in the document]
Comment (hosainnet) [reply steps/choose-a-response/ChooseAResponse.step.js]: Yep, it's fine. Are you refering to the line length?

File: steps/respond/Respond.step.js [hunk 7]
```diff
@@ -13,6 +13,11 @@ class Respond extends Interstitial {
     return this.req.session;
   }
 
+  handler(req, res) {
+    req.session.entryPoint = this.name;
```
[the remark this answers is not in the document]
Comment (hosainnet) [reply steps/respond/Respond.step.js]: CheckYourAnswers requires this to be set in the session

File: test/e2e/pages/checkYourAnswers.step.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+const content = require('steps/check-your-answers/CheckYourAnswers.content');
+
+function seeCheckYourAnswersPage() {
+  const I = this;
+
+  I.seeCurrentUrlEquals(CheckYourAnswersPage.path);
+  I.see(content.en.title);
+}
+
+function submit() {
```
[the remark this answers is not in the document]
Comment (hosainnet) [reply test/e2e/pages/checkYourAnswers.step.js]: 👍

File: steps/check-your-answers/CheckYourAnswers.html [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    sendApplication: content.submit
+} %}
+
+{% block statement_of_truth_content %}
+{{ header(content.confirm, size='medium') }}
+<p>{{ content.statementOfTruth | safe }}</p>
+{% endblock %}
+
+{% block statement_of_truth_fields %}
+    <div id="cya-confirmation-container">
```
[the remark this answers is not in the document]
Comment (hosainnet) [reply steps/check-your-answers/CheckYourAnswers.html]: 👍

File: package.json (hunk 2 above)
Comment (hosainnet) [reply package.json]: Ah yep good spot, thanks

File: steps/choose-a-response/ChooseAResponse.step.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (dtaylor7) [reply steps/choose-a-response/ChooseAResponse.step.js]: yea i was..

File: steps/respond/Respond.step.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (dtaylor7) [reply steps/respond/Respond.step.js]: no it shouldnt do.... if you remove whats the error?

[the remark this answers is not in the document]
Comment (dtaylor7) [reply steps/respond/Respond.step.js]: https://github.com/hmcts/div-decree-nisi-frontend/blob/master/steps/check-your-answers/CheckYourAnswers.step.js

[the remark this answers is not in the document]
Comment (hosainnet) [reply steps/respond/Respond.step.js]:
```
2018-09-18T09:43:06+01:00 - error: [one-per-page.errorPages] /check-your-answers message=No entryPoint defined in session, originalStack=Error: No entryPoint defined in session
    at AppInsightsAsyncCorrelatedErrorWrapper.ZoneAwareError (/Users/work/Dev/div-respondent-frontend/node_modules/zone.js/dist/zone-node.js:811:33)
    at new AppInsightsAsyncCorrelatedErrorWrapper (/Users/work/Dev/div-respondent-frontend/node_modules/applicationinsights/out/AutoCollection/CorrelationContextManager.js:193:18)
    at RequestBoundJourney.get entryPoint [as entryPoint] (/Users/work/Dev/div-respondent-frontend/node_modules/@hmcts/one-per-page/src/flow/RequestBoundJourney.js:81:11)
```
I'm guessing CYA needs to one from which points (i.e step) it needs to start checking the answers for?
 See https://github.com/hmcts/div-decree-nisi-frontend/blob/master/steps/undefended/Undefended.step.js#L17

[the remark this answers is not in the document]
Comment (dtaylor7) [reply steps/respond/Respond.step.js]: good call...

**Pull request merged**